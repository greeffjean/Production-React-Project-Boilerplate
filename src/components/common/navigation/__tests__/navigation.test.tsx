import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Navigation from "../navigation.component";
import { Provider } from "react-redux";
import store from "../../../../redux/store";
import { createMemoryHistory } from "history";
import {
  screen,
  waitFor,
  within,
  render,
  cleanup,
} from "@testing-library/react";

beforeAll(() => cleanup());

describe("Navigation tests", () => {
  const history = createMemoryHistory();

  const tree = renderer
    .create(
      <Router history={history}>
        <Provider store={store}>
          <Navigation />
        </Provider>
      </Router>
    )
    .toJSON();

  test("Matches snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  test("Contain correct amount of allowed routes", async () => {
    render(
      <Router history={history}>
        <Provider store={store}>
          <Navigation />
        </Provider>
      </Router>
    );

    const links = await waitFor(() => document.getElementsByTagName("a"));

    expect(links.length).toEqual(3);
  });
});
