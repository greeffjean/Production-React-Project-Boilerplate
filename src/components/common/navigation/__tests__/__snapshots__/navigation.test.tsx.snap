// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Matches snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        role="nav-list"
      >
        <a
          data-testid="link"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          data-testid="link"
          href="/dashboard-one"
          onClick={[Function]}
        >
          Dashboard One
        </a>
        <a
          data-testid="link"
          href="/dashboard-two"
          onClick={[Function]}
        >
          Dashboard Two
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Matches snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        role="nav-list"
      >
        <a
          data-testid="link"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          data-testid="link"
          href="/dashboard-one"
          onClick={[Function]}
        >
          Dashboard One
        </a>
        <a
          data-testid="link"
          href="/dashboard-two"
          onClick={[Function]}
        >
          Dashboard Two
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Navigation tests Matches snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        role="nav-list"
      >
        <a
          data-testid="link"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          data-testid="link"
          href="/dashboard-one"
          onClick={[Function]}
        >
          Dashboard One
        </a>
        <a
          data-testid="link"
          href="/dashboard-two"
          onClick={[Function]}
        >
          Dashboard Two
        </a>
      </div>
    </div>
  </header>
</div>
`;
